git_hooks/pre-commit

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+#
+# Ahook script to verify what is about to be committed, specifially wheter a flyway migration file matches the naming convention.
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+#
+
+if git rev-parse --verify HEAD >/dev/null 2>&1
+then
+	against=HEAD
+else
+	# Initial commit: diff against an empty tree object
+	against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
+fi
+
+# Redirect output to stderr.
+exec 1>&2
+
+naming_convention="(^.*[/][V][0-9]{17}[_]{2}.*\.sql$)"
```

Comment:
The regex looks like its more complicated than could be possible since it takes into account the path as part of the filename. Would it be possible to later-on use something like `$(basename $sql_file)` to discard the path making the pattern matching simpler?